microcosm_flask/tests/conventions/test_query.py

```diff
@@ -0,0 +1,140 @@
+"""
+Query operation tests.
+
+"""
+from json import loads
+from hamcrest import (
+    assert_that,
+    equal_to,
+    is_,
+)
+
+from marshmallow import fields, Schema
+from microcosm.api import create_object_graph
+
+from microcosm_flask.conventions.encoding import dump_response_data, load_query_string_data
+from microcosm_flask.conventions.registry import qs, response
+from microcosm_flask.operations import Operation
+from microcosm_flask.paging import PageSchema
+
+
+class QueryStringSchema(PageSchema):
+    value = fields.String(required=True)
+
+
+class QueryResultSchema(Schema):
+    result = fields.Boolean(required=True)
+    value = fields.String(required=True)
+
+
+def make_query(graph, request_schema, response_schema):
+    """
+    Create an example query route.
+
+    """
+
+    @graph.route("/v1/foo/get", Operation.Query, "foo")
+    @qs(request_schema)
+    @response(response_schema)
+    def foo_query():
+        """
+        My doc string
+        """
+        request_data = load_query_string_data(request_schema)
```

Comment:
The Search "operation" already exists and is part of the CRUD convention. It's the name I gave to a `GET` against the collection of resources and it can have query parameters -- by default it supports paging parameters, but this can be extended.

I think this is the same as a "list endpoint".